aspnetcore/fundamentals/configuration/index.md

```diff
@@ -26,13 +26,13 @@ Configuration in ASP.NET Core is performed using one or more [configuration prov
 * Directory files
 * In-memory .NET objects
 
-This topic provides information on configuration in ASP.NET Core. For information on using configuration in console apps, see [.NET Configuration](/dotnet/core/extensions/configuration).
+ASP.NET Core apps configure and launch a *host*. The host is responsible for app startup and lifetime management. The ASP.NET Core templates create a <xref:Microsoft.AspNetCore.Builder.WebApplicationBuilder> which contains the host. While some configuration can be done in both the host and the application configuration providers, generally, only configuration that is necessary for the host should be done in host configuration.
 
-[View or download sample code](https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/fundamentals/configuration/index/samples) ([how to download](xref:index#how-to-download-a-sample))
+This article provides information on configuration in ASP.NET Core. For information on using configuration in console apps, see [.NET Configuration](/dotnet/core/extensions/configuration).
 
 <a name="default"></a>
 
-## Default configuration
+## Default application configuration
```

Comment:
I think the structure should be as follows.

```
## Default configuration
### Application configuration
### Host configuration
### Host bootstrap settings
```

I also think that the section discussing default application config should be reordered to go from highest to lowest priority along with the same list for default host configuration sources. As ordered right now, the highest priority source for application config (Command-line arguments) is in the middle of the combined list of application and host sources which makes no sense. Application config falls back to host config, so I want to be able to skim down the list of sources without the order resetting in the middle.

The following should be deleted:

> 1. [ChainedConfigurationProvider](xref:Microsoft.Extensions.Configuration.ChainedConfigurationSource) : Adds an existing `IConfiguration` as a source. In the default configuration case, adds the [host](#hvac) configuration and setting it as the first source for the *app* configuration.

It's super confusing. Hardly anyone should care about the mechanism that application config falls back to hosting config (which isn't entirely accurate anymore anyway for `WebApplicationBuilder`). I think it similar to the `user secrets` where the actual configuration provider used to make this fallback work (if any) is unimportant.

The last (lowest-priority) element of the list of default application sources should just read:

> 1. A fallback to the host configuration described in the next section

Can we also update `[App secrets](xref:security/app-secrets)` to `[user-secrets](xref:security/app-secrets)` and `Environment variables` to `Non-prefixed environment variables` similar to in my comment?